src/oskbd/macos.rs

```
@@ -0,0 +1,230 @@
+            })
+        } else {
+            Err(())
+        }
+    }
+}
+
+pub struct KbdOut {}
+
+impl KbdOut {
+    pub fn new() -> Result<Self, io::Error> {
+        Ok(KbdOut {})
+    }
+
+    pub fn write(&mut self, event: InputEvent) -> Result<(), io::Error> {
+        let mut devent = event.into();
+        log::debug!("Attempting to write {event:?} {devent:?}");
+        let _sent = send_key(&mut devent);
+        Ok(())
+    }
+
+    pub fn write_key(&mut self, key: OsCode, value: KeyValue) -> Result<(), io::Error> {
+        if let Ok(event) = InputEvent::try_from(KeyEvent { value, code: key }) {
+            self.write(event)
+        } else {
+            log::debug!("couldn't write unrecognized {key:?}");
+            Err(io::Error::new(
+                io::ErrorKind::NotFound,
+                "OsCode not recognized!",
+            ))
+        }
+    }
+
+    pub fn write_code(&mut self, code: u32, value: KeyValue) -> Result<(), io::Error> {
+        if let Ok(event) = InputEvent::try_from(KeyEvent {
+            value,
+            code: OsCode::from_u16(code as u16).unwrap(),
+        }) {
+            self.write(event)
+        } else {
+            log::debug!("couldn't write unrecognized OsCode {code}");
+            Err(io::Error::new(
+                io::ErrorKind::NotFound,
+                "OsCode not recognized!",
+            ))
+        }
+    }
+
+    pub fn press_key(&mut self, key: OsCode) -> Result<(), io::Error> {
+        self.write_key(key, KeyValue::Press)
+    }
+
+    pub fn release_key(&mut self, key: OsCode) -> Result<(), io::Error> {
+        self.write_key(key, KeyValue::Release)
+    }
+
+    /// Send using C-S-u + <unicode hex number> + spc
+    pub fn send_unicode(&mut self, c: char) -> Result<(), io::Error> {
+        log::debug!("sending unicode {c}");
+        todo!();
+    }
+
+    pub fn scroll(&mut self, _direction: MWheelDirection, _distance: u16) -> Result<(), io::Error> {
+        Ok(())
```

Please add one of panic/todo/error log to state that these are unsupported in macos for the time being.